### ISSUE TITLE
Can't acces function static properties if function name is same as script name

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#5032]
### Description
Since `2024.400` (first beta) I can't anymore access static properties of non-constructor (initialized) function anymore, if it's name is same as script asset name.

### Steps To Reproduce
1. Create script
2. Name script as `B`
3. Put below code:
```gml
function A() {
  static bar = function() {};
  static val = 1;
}
A();
A.bar(); // works
A.val = 1; // works

// comment code below, and game would compile
function B() {
  static bar = function() {};
  static val = 1;
}
B();
B.bar(); // compile error (only if script name == B)
B.val = 1; // compile error (only if script name == B)
```
4. run. there would be errors:
```
Script: B at line 15 : cannot use function/script name for a variable, using "B"
Script: B at line 15 : malformed assignment
```
5. comment lines 10-16
6. run
7. there's no compile errors for script `B` if function `B` is commented

### Which version of GameMaker are you reporting this issue for?
IDE v2024.400.0.516 Runtime v2024.400.0.537

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

[75d4fd7a-4f8f-4b24-97e7-01cacae99977](https://api.gamemaker.io/api/github/downloads/75d4fd7a-4f8f-4b24-97e7-01cacae99977)

**Comment by rwkay** [comment]
I need to think about this one as the Script asset reference has priority over the function name just now... though arguably in this context it is the function name that should be resolved...

**Comment by gnysek** [comment]
Ok, so seems that change was to fix non-intuitive solution for what `script_execute` would do since version 2.3+, when there was function/script name collision (from what I remember - it would call function instead of script body, and to call "script" only way was to deduct it's reference doing loop from 100000+, and checking if `string(method(self, _s));` contains `function gml_Script_`).

Would be bad if `Func.property` added back in 2023.1 / 2023.2 would not work anymore.

I would of course propose new operator here (like `func::property`), but I understand that GMLv2 is not getting new features anymore, so that's rather not a solution (and would require not only runtime support, but also IDE support for both code editors).
Another solution could be adding `script_execute_with_body(<ref_script> script);` or sth like that, so compiler would throw error, if function name is given there, but there's no script like that in project.

For now it seems that best solution is to (yet) not change anything.

**Comment by rwkay** [comment]
Fixed in 2024.4

**Comment by Emc1923** [comment]
Verified fixed as of  IDE v2024.400.0.533  Runtime v2024.400.0.552
